classes/RightSignature/HttpClient.php

```diff
@@ -2,81 +2,154 @@
+     */
+	public static function forToken($apiToken, ClientInterface $client = null)
 	{
-		$instance = new self();
-		$instance->_client->addHeader(sprintf('Api-Token: %s', $apiToken));
+		if (! $client) {
+			$client = new Client([
+				'base_uri' => \RightSignature::API_ENDPOINT,
+			]);
+		}
+		$instance = new self($client);
+		$instance->setToken($apiToken);
+
 		return $instance;
 	}
 
-	public function __construct()
+	/**
+	 * @param \GuzzleHttp\ClientInterface $client
+     */
+	private function __construct(ClientInterface $client)
 	{
-		$this->_client = new \Ergo\Http\Client(\RightSignature::API_ENDPOINT);
-		$this->_client->addHeader(sprintf('Api-Version: %s', \RightSignature::API_VERSION));
+		$this->_client = $client;
 	}
 
 	/**
 	 * @param string $path
 	 * @return string
-	 * @throws RightSignature\Exception
+	 * @throws Exception
 	 */
 	public function get($path)
 	{
-		$response = $this->_submit('get', $path);
-		return $response;
+		$getRequest = [
+			$path, [
+				'headers' => [
+					'Api-Token' => $this->_apiToken,
+					'Api-Version' => \RightSignature::API_VERSION
+				],
+				'verify' => false
+			]
+		];
+
+		return $this->_submit('get', $getRequest);
 	}
 
 	/**
 	 * @param string $path
 	 * @param string $body
 	 * @return string
-	 * @throws RightSignature\Exception
+	 * @throws Exception
 	 */
 	public function post($path, $body=null)
 	{
-		$response = $this->_submit('post', $path, $body, $body ? 'application/xml' : null);
-		return $response;
-	}
+		$postRequest = [
+			$path, [
+				'headers' => [
+					'Api-Token' => $this->_apiToken,
+					'Api-Version' => \RightSignature::API_VERSION
+				],
+				'verify' => false
+			]
+		];
 
-	// ----------------------------------------
-	// Private methods
+		if (is_array($body))
+		{
+			$postRequest['form_params'] = $body;
+		}
+		else
+		{
+			$postRequest['body'] = $body;
+		}
 
-	private function _submit($method /*, $args...*/)
-	{
-		$args = array_slice(func_get_args(), 1);
+		return $this->_submit('post', $postRequest);
+	}
+
+	/**
+	 * @param $method
+	 * @param array $params
+	 * @return mixed
+	 * @throws Exception
+     */
+	private function _submit($method, Array $params) {
 		try
 		{
-			$response = call_user_func_array(array($this->_client, $method), $args);
-			return $response->getBody();
+			$response = call_user_func_array(array($this->_client, $method), $params);
+			if ($response->getStatusCode() === 200) {
+				return $response->getBody()->getContents();
+			}
+
+			// Or throw an exception
+			throw self::_translateError($response->getStatusCode(), $response->getBody()->getContents());
 		}
-		catch (\Ergo\Http\Error $e)
+		catch (RequestException $ex)
 		{
-			throw self::_translateError($e);
+			if ($ex->hasResponse())
+			{
+				$response = $ex->getResponse();
+				throw self::_translateError($response->getStatusCode(), $response->getBody()->getContents());
+			}
+
+			// Default to a 400 error
+			throw self::_translateError(400);
```

Comment:
This would raise an exception with status 400 and message `Internal Server Error`. Could we make the second argument to `_translateError()` non-optional?